servers/src/main/java/tachyon/master/block/BlockMaster.java

```diff
@@ -149,14 +150,22 @@ public BlockMaster(Journal journal) {
   }
 
   @Override
-  public TProcessor getProcessor() {
-    return new BlockMasterService.Processor<BlockMasterServiceHandler>(
-        new BlockMasterServiceHandler(this));
+  public Map<String, TProcessor> getServices() {
+    Map<String, TProcessor> services = new HashMap<String, TProcessor>();
+    services.put(
+        Constants.BLOCK_MASTER_CLIENT_SERVICE_NAME,
+        new BlockMasterClientService.Processor<BlockMasterClientServiceHandler>(
```

Comment:
This method is invoked exactly only once when the Tachyon master starts up so there is no need for caching these.